src/main/java/org/apache/ibatis/builder/annotation/MapperAnnotationBuilder.java

```diff
@@ -333,7 +333,9 @@ void parseStatement(Method method) {
         fetchSize = options.fetchSize() > -1 || options.fetchSize() == Integer.MIN_VALUE ? options.fetchSize() : null; //issue #348
         timeout = options.timeout() > -1 ? options.timeout() : null;
         statementType = options.statementType();
-        resultSetType = options.resultSetType();
+        if (options.resultSetType() != ResultSetType.DEFAULT) {
+          resultSetType = options.resultSetType();
+        }
```

Comment:
@harawata Thanks for your review and sorry for late replying...

I've fixed via d93ae87b3fe5d9e58fd23f20e4757876d6f37813. 
I've add `NONE` as new enum value for resolving your review comment. It is value that indicate to not set and it is design to use as default value on `@Options#resultSetType()`.

Reply:
Oh, I thought this PR was mergeable already.
It may not make sense for a user to specify `@Options(resultSetType=DEFAULT)`, right?
I am really sorry that I wasn't very clear, but I think `ResultSetType.NONE` is unnecessary...

Reply:
@harawata Oh...I've mistaken reading your comment. I've revert previous commit and applied some small changed.